sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -248,6 +263,9 @@ class __SYCL_EXPORT modifiable_command_graph {
   /// @return Node added to the graph.
   node addImpl(const std::vector<node> &Dep);
 
+  // Adds all graph leaves as dependencies to the current node
+  void dependsOnLeaves(node Node);
```

Comment:
```suggestion
  void dependOnAllLeaves(node Node);
```
Minor change but I think being more specific is good and in this context `depends` sounds a bit more like a boolean query than a function that's actually doing something.

Reply:
This name has already been updated in the meantime to `addGraphLeafDependencies` (according to Ewan's suggestion)